### PR TITLE
Add replication support

### PR DESCRIPTION
**Pull request by pessoa**
This allows to optionally configure mysql/mariadb replication monitoring.

### CODE REVIEW

**Comment by carlosperello** [comment]
👍

File: manifests/plugin/mysql.pp [hunk 1]
```diff
@@ -42,6 +46,7 @@
   $pass = undef,
   $port = undef,
   $sock = undef,
+  $repl = False,
```
Comment (carlosperello) [inline manifests/plugin/mysql.pp]: shouldn't this be just false ? https://puppet.com/docs/puppet/5.3/lang_data_boolean.html

File: templates/plugin/mysql.yaml.erb [hunk 2]
```diff
@@ -13,6 +13,10 @@ instances:
 <%- end -%>
 <%- if @sock -%>
     sock: <%= @sock %>
+<%- if @repl -%>
```
Comment (shurl) [inline templates/plugin/mysql.yaml.erb]: @pessoa This is inside the if statement for sock. So you can only enable replication if you set the socket. I presume this isn't intended behaviour?

**Comment by shurl** [comment]
LGTM!

**Pull request merged**